pkg/ring/consul_client.go

```diff
@@ -203,6 +210,54 @@ func (c *consulClient) WatchKey(ctx context.Context, key string, f func(interfac
 	}
 }
 
+// WatchPrefix will watch a given prefix in Consul for new keys and changes to existing keys under that prefix.
+// When the value under said key changes, the f callback is called with the deserialised value.
+// To construct the deserialised value, a factory function should be supplied which generates an empty
```

Comment:
> To construct the deserialised value, a factory function should be supplied which generates an empty struct for WatchKey to deserialise into.

This statement can be omitted. It's not passed in this function and only stands to confuse the user.